src/Microsoft.IdentityModel.Logging/TextWriterEventListener.cs

```diff
@@ -0,0 +1,101 @@
+﻿//-----------------------------------------------------------------------
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved
+// Apache License 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Event listener that writes logs to a file or a fileStream provided by user.
+    /// </summary>
+    public class TextWriterEventListener : EventListener
+    {
+        private StreamWriter _streamWriter;
+        private bool _disposeStreamWriter = true;
+
+        public readonly static string DefaultLogFileName = "IdentityModelLogs.txt";
+
+        public TextWriterEventListener()
+        {
+             Stream fileStream = new FileStream(DefaultLogFileName, FileMode.OpenOrCreate, FileAccess.Write);
```

Comment:
It will throw exception.: System.UnauthorizedAccessException (when file cannot be written) or System.IO.IOException (when it cannot create new).